examples/manipulation/grasp_eval.py

```diff
@@ -19,73 +22,234 @@
+    stereo_img = get_stereo_frame(env, step_count)
+
+    # Display combined image
+    cv2.imshow("Stereo Cameras", stereo_img)
+
+    # Wait for key press (1ms delay)
+    key = cv2.waitKey(1)
+    if key == 27:  # ESC key
+        return False
+    return True
+
+
+def save_frames_as_video(frames, video_path, fps=60):
+    """Save a list of frames as a video."""
+    if not frames:
+        print("No frames to save!")
+        return
+
+    # Get frame dimensions from the first frame
+    height, width = frames[0].shape[:2]
+
+    # Create video writer
+    fourcc = cv2.VideoWriter_fourcc(*"mp4v")
+    video_writer = cv2.VideoWriter(video_path, fourcc, fps, (width, height))
+
+    if not video_writer.isOpened():
+        print(f"Error: Could not open video writer for {video_path}")
+        return
+
+    # Write frames to video
+    print(f"Saving {len(frames)} frames to {video_path}...")
+    for i, frame in enumerate(frames):
+        video_writer.write(frame)
+        if i % 30 == 0:  # Progress indicator every 30 frames
+            print(f"Progress: {i + 1}/{len(frames)} frames")
+
+    # Release video writer
+    video_writer.release()
+    print(f"Video saved successfully: {video_path}")
 
 
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("-e", "--exp_name", type=str, default="grasp")
-    parser.add_argument("--record", action="store_true", default=False)
+    parser.add_argument(
+        "--stage",
+        type=str,
+        default="rl",
+        choices=["rl", "bc"],
+        help="Model type: 'rl' for reinforcement learning, 'bc' for behavior cloning",
+    )
+    parser.add_argument(
+        "--record",
+        action="store_true",
+        help="Record stereo images as video during evaluation",
+    )
+    parser.add_argument(
+        "--video_path",
+        type=str,
+        default=None,
+        help="Path to save the video file (default: auto-generated)",
+    )
     args = parser.parse_args()
 
+    # Set PyTorch default dtype to float32 for better performance
+    torch.set_default_dtype(torch.float32)
+
     gs.init()
 
-    log_dir = f"logs/{args.exp_name}"
-    last_folder = sorted(os.listdir(log_dir))[-1]
-    env_cfg, reward_cfg, robot_cfg, train_cfg = pickle.load(open(f"logs/{args.exp_name}/{last_folder}/cfgs.pkl", "rb"))
+    log_dir = f"logs/{args.exp_name + '_' + args.stage}"
+
+    # Load configurations
+    if args.stage == "rl":
+        # For RL, load the standard configs
+        env_cfg, reward_cfg, robot_cfg, rl_train_cfg, bc_train_cfg = pickle.load(open(f"{log_dir}/cfgs.pkl", "rb"))
+    else:
+        # For BC, we need to load the configs and create BC config
+        env_cfg, reward_cfg, robot_cfg, rl_train_cfg, bc_train_cfg = pickle.load(open(f"{log_dir}/cfgs.pkl", "rb"))
 
-    # visualize the target
-    env_cfg["visualize_target"] = True
-    # for video recording
-    env_cfg["visualize_camera"] = args.record
```

Comment:
Why removing these options ?

Reply:
This is no longer the case and you can now I have both, and it is fully unit tested. So I would suggest to restore these options :)

Reply:
this is not done yet. I will address it tomorrow. have to go home now.

Reply:
I think this bug should be fixed before merging. I will work on it.

Reply:
I just realised the issue ! you forgot to specify `debug=True` for your `vis_cam` !!

Reply:
I see. Updated my script. Now, it uses the default camera rescoring, similar to the previous examples.